libs/awscdk/src/on-deploy.ts

```diff
@@ -17,7 +17,7 @@ export class OnDeploy extends cloud.OnDeploy {
   ) {
     super(scope, id, handler, props);
 
-    let fn = new cloud.Function(this, "Function", handler, props);
+    let fn = new cloud.Function(this, "Function", handler as cloud.IFunctionHandler, props);
```

Comment:
Is this ok?

Reply:
Yeah, structurally the types match. typescript would yell otherwise, and we'd have to use our `convertBetweenHandlers` function. 
I'm guessing it wasn't needed before because `handler`'s type was just an empty interface that extended IInflight.